library/src/main/java/org/apache/apex/malhar/lib/fs/S3RecordReader.java

```diff
@@ -0,0 +1,541 @@
+      }
+      byte[] record = Arrays.copyOfRange(buffer, Long.valueOf(bufferOffset).intValue(),
+          Long.valueOf(bufferOffset + bufferLength).intValue());
+      bufferOffset += record.length;
+      entity.setRecord(record);
+      entity.setUsedBytes(record.length);
+      return entity;
+    }
+
+    /**
+     * Set the AmazonS3 service
+     *
+     * @param s3Client
+     *          given s3Client
+     */
+    public void setS3Client(AmazonS3 s3Client)
+    {
+      this.s3Client = s3Client;
+    }
+
+    /**
+     * Set the bucket name
+     *
+     * @param bucketName
+     *          given bucketName
+     */
+    public void setBucketName(String bucketName)
+    {
+      this.bucketName = bucketName;
+    }
+
+    /**
+     * Sets the file path
+     *
+     * @param filePath
+     *          given filePath
+     */
+    public void setFilePath(String filePath)
+    {
+      this.filePath = filePath;
+    }
+
+    /**
+     * @param fileLength
+     *          length of the file to which the block belongs
+     */
+    public void setFileLength(long fileLength)
+    {
+      this.fileLength = fileLength;
+    }
+
+    /**
+     * Sets the offset within the current buffer
+     *
+     * @param bufferOffset
+     *          offset within the current buffer
+     */
+    protected void setBufferOffset(int bufferOffset)
+    {
+      this.bufferOffset = bufferOffset;
+    }
+  }
+
+  /**
+   * Size of bytes to be retrieved when a record overflows
+   *
+   * return overflowBufferSize
+   */
+  public int getOverflowBufferSize()
+  {
+    return overflowBufferSize;
+  }
+
+  /**
+   * Size of bytes to be retrieved when a record overflows
+   *
+   * @param overflowBufferSize
+   */
+  public void setOverflowBufferSize(int overflowBufferSize)
+  {
+    this.overflowBufferSize = overflowBufferSize;
+  }
+
+  /**
+   * Get the S3 bucket name
+   *
+   * @return bucket
+   */
+  public String getBucketName()
+  {
+    return bucketName;
+  }
+
+  /**
+   * Set the bucket name where the file resides
+   *
+   * @param bucketName
+   *          bucket name
+   */
+  public void setBucketName(String bucketName)
+  {
+    this.bucketName = bucketName;
+  }
+
+  /**
+   * Return the access key
+   *
+   * @return the accessKey
+   */
+  public String getAccessKey()
+  {
+    return accessKey;
+  }
+
+  /**
+   * Set the access key
+   *
+   * @param accessKey
+   *          given accessKey
+   */
+  public void setAccessKey(String accessKey)
+  {
+    this.accessKey = accessKey;
+  }
+
+  /**
+   * Return the secretAccessKey
+   *
+   * @return the secretAccessKey
+   */
+  public String getSecretAccessKey()
+  {
+    return secretAccessKey;
+  }
+
+  /**
+   * Set the secretAccessKey
+   *
+   * @param secretAccessKey
+   *          secretAccessKey
+   */
+  public void setSecretAccessKey(String secretAccessKey)
+  {
+    this.secretAccessKey = secretAccessKey;
+  }
+
+  /**
+   * S3 endpoint
+   *
+   * @param s3
+   *          endpoint
```

Comment:
I think there is no parameter "s3" in setter method. Please verify and update it.